portal/app/[locale]/tunnel/_hooks/useWaitTime.ts

```diff
@@ -0,0 +1,24 @@
+import { useQuery } from '@tanstack/react-query'
+import { ToEvmWithdrawOperation } from 'types/tunnel'
+import {
+  getTimeToProveInSeconds,
+  getTimeToFinalizeInSeconds,
+} from 'utils/wait-times/evmWithdrawals'
+
+export const useEvmWithdrawTimeToProve = (withdrawal: ToEvmWithdrawOperation) =>
+  useQuery({
+    enabled: !!withdrawal,
```

Comment:
If the withdrawal is defined, this will query the `getTimeToProveInSeconds` despite the status of the withdrawal. But it's a call that's only meaningful in certain states. We could improve this (and `useEvmWithdrawTimeToFinalize`) by running the query when we actually need to. Something like

```suggestion
    enabled: !!withdrawal && [MessageStatus.STATE_ROOT_NOT_PUBLISHED, MessageStatus.READY_TO_PROVE].includes(withdrawal.status),
```

I think we could even get rid of `READY_TO_PROVE`, but let's be defensive and keep it.

Similar in line 17, but with `IN_CHALLENGE_PERIOD` and  `READY_FOR_RELAY` .


Note that `useQuery` will return `undefined` when disabled (or a cached value if any), so you may need to handle those scenarios.


With the current implementation, if a withdrawal is ready to claim, it is still calling the `getTimeToProveInSeconds`, which we know for a fact that won't be rendered, as at this stage, the wait should be completed (otherwise, it wouldn't be ready for claim!)